partclone does not compile with uclibc due to mcheck.h header missing

We use the cross-compiler buildroot to compile a project integrating partclone. Builroot uses the uclibc library by default instead of glibc.
The following error is displayed during partclone compilation:

main.c:25:10: fatal error: mcheck.h: No such file or directory
   25 | #include <mcheck.h>
      |          ^~~~~~~~~~
Indeed mcheck.h is not part of uclibc whereas it is available on glibc. 

I think the mcheck header should be included in a MEMTRACE ifdef the same way as the mtrace call. 
Ideally configure should not allow enable-memtrace if mcheck.h header does not exist.

Also it does not seem the -DMEMTRACE option is  properly used when compiling main.c as after patching main.c to only include mcheck.h within a #ifdef MEMTRACE, it compiles even if enable-memtrace is set whereas I expected the same error message.

--- a/config.h.in
+++ b/config.h.in
@@ -79,6 +79,9 @@
 /* Define to 1 if you have the <mqueue.h> header file. */
 #undef HAVE_MQUEUE_H
 
+/* Define to 1 if you have the <mtrace.h> header file. */
+#undef HAVE_MTRACE_H
+
 /* Define to 1 if you have the <ncurses.h> header file. */
 #undef HAVE_NCURSES_H
 

--- a/configure.ac
+++ b/configure.ac
@@ -457,6 +457,10 @@ AC_ARG_ENABLE([mtrace],
         [--enable-mtrace],
         [enable memory tracing])
 )
+if test "$enable_mtrace" = "yes"; then
+AC_CHECK_HEADERS([mtrace.h], ,
+    AC_MSG_ERROR([*** mtrace header file (mtrace.h) not found]))
+fi
 enable_memtrace=$enable_mtrace
 AM_CONDITIONAL(ENABLE_MEMTRACE, test "$enable_memtrace" = yes)
 

--- a/src/main.c
+++ b/src/main.c
@@ -26,7 +26,9 @@
 #include <sys/types.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef MEMTRACE
 #include <mcheck.h>
+#endif
 #include <stdint.h>
 #include <stdarg.h>
 #include <string.h>

